superset/assets/src/dashboard/components/filterscope/FilterScopeSelector.jsx

```diff
@@ -0,0 +1,518 @@
+      checkedFilterFields,
+      filterScopeMap,
+    } = this.state;
+    const key = getKeyForFilterScopeTree({
+      activeFilterField,
+      checkedFilterFields,
+    });
+    const updatedEntry = {
+      ...filterScopeMap[key],
+      expanded,
+    };
+    this.setState(() => ({
+      filterScopeMap: {
+        ...filterScopeMap,
+        [key]: updatedEntry,
+      },
+    }));
+  }
+
+  onCheckFilterField(checkedFilterFields = []) {
+    const { layout } = this.props;
+    const { filterScopeMap } = this.state;
+    const filterScopeTreeEntry = buildFilterScopeTreeEntry({
+      checkedFilterFields,
+      activeFilterField: '',
+      filterScopeMap,
+      layout,
+    });
+
+    this.setState(() => ({
+      activeFilterField: '',
+      checkedFilterFields,
+      filterScopeMap: {
+        ...filterScopeMap,
+        ...filterScopeTreeEntry,
+      },
+    }));
+  }
+
+  onExpandFilterField(expandedFilterIds = []) {
+    this.setState(() => ({
+      expandedFilterIds,
+    }));
+  }
+
+  onChangeFilterField(filterField = {}) {
+    const { layout } = this.props;
+    const nextActiveFilterField = filterField.value;
+    const {
+      activeFilterField: currentActiveFilterField,
+      checkedFilterFields,
+      filterScopeMap,
+    } = this.state;
+
+    // we allow single edit and multiple edit in the same view.
+    // if user click on the single filter field,
+    // will show filter scope for the single field.
+    // if user click on the same filter filed again,
+    // will toggle off the single filter field,
+    // and allow multi-edit all checked filter fields.
+    if (nextActiveFilterField === currentActiveFilterField) {
+      const filterScopeTreeEntry = buildFilterScopeTreeEntry({
+        checkedFilterFields,
+        activeFilterField: '',
+        filterScopeMap,
+        layout,
+      });
+
+      this.setState({
+        activeFilterField: '',
+        filterScopeMap: {
+          ...filterScopeMap,
+          ...filterScopeTreeEntry,
+        },
+      });
+    } else if (this.allfilterFields.includes(nextActiveFilterField)) {
+      const filterScopeTreeEntry = buildFilterScopeTreeEntry({
+        checkedFilterFields,
+        activeFilterField: nextActiveFilterField,
+        filterScopeMap,
+        layout,
+      });
+
+      this.setState({
+        activeFilterField: nextActiveFilterField,
+        filterScopeMap: {
+          ...filterScopeMap,
+          ...filterScopeTreeEntry,
+        },
+      });
+    }
+  }
+
+  onSearchInputChange(e) {
+    this.setState({ searchText: e.target.value }, this.filterTree);
+  }
+
+  onClose() {
+    this.props.onCloseModal();
+  }
+
+  onSave() {
+    const { filterScopeMap } = this.state;
+
+    console.log(
+      'i am current state',
+      this.allfilterFields.reduce(
+        (map, key) => ({
+          ...map,
+          [key]: filterScopeMap[key].checked,
+        }),
+        {},
+      ),
+    );
+
+    // save do not close modal
```

Comment:
grammar nit: `save does not close modal`

Reply:
fixed.